root/static/scripts/edit/URLCleanup.js

```diff
@@ -321,12 +322,13 @@ const CLEANUPS = {
     type: LINK_TYPES.discogs,
     clean: function (url) {
       url = url.replace(/\/viewimages\?release=([0-9]*)/, "/release/$1");
-      url = url.replace(/^https?:\/\/(?:[^.]+\.)?discogs\.com\/(?:.*\/)?(user\/[^\/#?]+|(?:artist|release|master(?:\/view)?|label)\/[0-9]+)(?:[\/#?-].*)?$/, "https://www.discogs.com/$1");
+      url = url.replace(/\/composition\/([^-]+-[^-]+-[^-]+-[^-]+-[^-]+).*/, "/composition/$1");
+      url = url.replace(/^https?:\/\/(?:[^.]+\.)?discogs\.com\/(?:.*\/)?(user|composition\/[^\/#?]+|(?:artist|release|master(?:\/view)?|label)\/[0-9]+)(?:[\/#?-].*)?$/, "https://www.discogs.com/$1");
       url = url.replace(/^(https:\/\/www\.discogs\.com\/master)\/view\/([0-9]+)$/, "$1/$2");
       return url;
     },
     validate: function (url, id) {
-      var m = /^https:\/\/www\.discogs\.com\/(?:(artist|label|master|release)\/[1-9][0-9]*|(user)\/.+)$/.exec(url);
+      var m = /^https:\/\/www\.discogs\.com\/(?:(artist|label|master|release)\/[1-9][0-9]*|(user|composition)\/.+)$/.exec(url);
```

Comment:
Validation can be improved by checking that the expected UUID after composition is complete.

Reply:
Okay--my new solution checks for 4 seperate "chunks" after /composition/, which means that it's a valid UUID. This *does* mean that I also need to check for m[3], which is unlike any other test case.